coordinator/src/main/scala/filodb.coordinator/RowSource.scala

```diff
@@ -23,24 +24,20 @@ object RowSource {
  *
  * To start initialization and reading from source, send the Start message.
  *
- * Backpressure and at least once mechanism:  RowSource sends rows to the NodeCoordinator,
- * but does not send more than maxUnackedRows before getting an ack back.  As long as it receives
- * acks it will keep sending, but will stop once maxUnackedRows is reached.  It then goes into a waiting
- * state, waiting for Acks to come back, or periodically pinging with a CheckCanIngest message.  If it
- * is allowed to ingest again, then it proceeds but rewinds from the last acked message.  This ensures
- * that we do not skip any incoming messages.
+ * Backpressure and at least once mechanism:  RowSource sends batches of rows to the NodeCoordinator,
+ * but does not send more than maxUnackedBatches before getting an ack back.  As long as it receives
+ * acks it will keep sending, but will stop once maxUnackedBatches is reached.  It then goes into a waiting
+ * state, waiting for Acks to come back.  Once waitingPeriod has elapsed, if it is allowed to ingest again,
+ * then it will replay unacked messages, hopefully get acks back, and once all messages are acked will go
+ * back into regular ingestion mode.  This ensures that we do not skip any incoming messages.
  */
 trait RowSource extends Actor with StrictLogging {
   import RowSource._
 
```

Comment:
Not part if this change set but in looking at the raw source I'd suggest this change:

     private var whoStartedMe: Option[ActorRef] = None

Then: whichever makes sense
 
     for (identity <- whoStartedMe) identity ! AllDone
     for (identity <- whoStartedMe) identity = sender
     whoStartedMe  map (_ ! AllDone)
     whoStartedMe  map (_ = sender)